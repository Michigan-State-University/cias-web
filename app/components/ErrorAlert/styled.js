import styled from 'styled-components';

export const StyledAlert = styled.div`
  padding: 20px;
  width: 100%;
  background-color: rgba(255, 0, 0, 0.2);
  border: 1px solid red;
  border-radius: 10px;
  margin: 10px 0;
  max-width: 700px;
`;
