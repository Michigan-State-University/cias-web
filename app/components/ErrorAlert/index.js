/**
 *
 * ErrorAlert
 *
 */

import React, { memo } from 'react';
import PropTypes from 'prop-types';
import identity from 'lodash/identity';

import Row from 'components/Row';
import Img from 'components/Img';
import Text from 'components/Text';
import exclamationMark from 'assets/svg/exclamationMark.svg';

import { AlertContainer } from './styled';

const ErrorAlert = ({ errorText, fullPage, ...restProps }) => {
  const toDisplay = errorText.toString().split('\n')[0];

  const wrapWithRow = child => (
    <Row width="100%" justify="center" mt={100}>
      {child}
    </Row>
  );

  const wrapper = fullPage ? wrapWithRow : identity;
  return wrapper(
    <AlertContainer {...restProps}>
      <Img src={exclamationMark} alt="error" mr={15} />
      <Text fontWeight="bold" fontSize={15}>
        {toDisplay}
      </Text>
    </AlertContainer>,
  );
};

ErrorAlert.propTypes = {
  errorText: PropTypes.node.isRequired,
  fullPage: PropTypes.bool,
};

export default memo(ErrorAlert);
