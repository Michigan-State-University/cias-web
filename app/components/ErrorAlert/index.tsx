/**
 *
 * ErrorAlert
 *
 */

import React from 'react';

import Row from 'components/Row';
import Img from 'components/Img';
import Text from 'components/Text';
import ConditionalWrapper from 'components/ConditionalWrapper';
import exclamationMark from 'assets/svg/exclamationMark.svg';

import { AlertContainer } from './styled';

type Props = {
  errorText: string | object;
  fullPage?: boolean;
} & Record<string, any>;

const ErrorAlert = ({ errorText, fullPage, ...restProps }: Props) => {
  const toDisplay = errorText.toString().split('\n')[0];

  return (
    <ConditionalWrapper
      if={fullPage}
      with={Row}
      wrapperProps={{ width: '100%', justify: 'center', mt: 100 }}
    >
      <AlertContainer {...restProps}>
        <Img src={exclamationMark} alt="error" mr={15} />
        <Text fontWeight="bold" fontSize={15}>
          {toDisplay}
        </Text>
      </AlertContainer>
    </ConditionalWrapper>
  );
};

export default ErrorAlert;
