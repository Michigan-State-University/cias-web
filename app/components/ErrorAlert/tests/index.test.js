/**
 *
 * Tests for ErrorAlert
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import ErrorAlert from '../index';

describe('<ErrorAlert />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<ErrorAlert errorText="error" />);
    expect(spy).not.toHaveBeenCalled();
  });

  /**
   * Unskip this test to use it
   *
   * @see {@link https://jestjs.io/docs/en/api#testskipname-fn}
   */
  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<ErrorAlert errorText="error" />);
    expect(firstChild).toMatchSnapshot();
  });
});
