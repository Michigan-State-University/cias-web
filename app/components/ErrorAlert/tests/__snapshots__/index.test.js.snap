// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorAlert /> Should render and match the snapshot 1`] = `
<div
  class="styled__StyledAlert-mky144-0 iMEfMz"
>
  error
</div>
`;
